File: node/src/commands/BitmapIndexType.ts
/**
 * Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
 */

// Imports below added to fix up the TSdoc links, but eslint complains about unused imports.
/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
import { BaseClient } from "src/BaseClient";
/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
import { BitOffsetOptions } from "src/commands/BitOffsetOptions";

/**
 * Enumeration specifying if index arguments are BYTE indexes or BIT indexes.
 * Can be specified in {@link BitOffsetOptions}, which is an optional argument to the {@link BaseClient.bitcount|bitcount} command.
 * Can also be specified as an optional argument to the {@link BaseClient.bitposInverval|bitposInterval} command.
 *
 * since - Valkey version 7.0.0.
 */
export enum BitmapIndexType {
    /** Specifies that provided indexes are byte indexes. */
    BYTE = "BYTE",
    /** Specifies that provided indexes are bit indexes. */
    BIT = "BIT",
}

File: java-old/client/src/main/java/glide/api/models/commands/bitmap/BitmapIndexType.java
/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
package glide.api.models.commands.bitmap;

import glide.api.commands.BitmapBaseCommands;

/**
 * Optional arguments for {@link BitmapBaseCommands#bitcount(String, long, long, BitmapIndexType)}.
 * Specifies if start and end arguments are BYTE indices or BIT indices
 *
 * @see <a href="https://valkey.io/commands/bitcount/">valkey.io</a>
 */
public enum BitmapIndexType {
    /** Specifies a byte index * */
    BYTE,
    /** Specifies a bit index */
    BIT
}
